native/src/main/java/io/ballerina/lib/data/xmldata/utils/DataUtils.java

```diff
@@ -465,10 +468,23 @@ public static Object getModifiedRecord(BMap<BString, Object> input, BString text
         return input;
     }
 
+    private static BMap<BString, Object> mergeOriginalAndCurrentAnnotations(
+                                BMap<BString, Object> originalTypeAnnotations, BMap<BString, Object> typeAnnotations) {
+        BMap<BString, Object> annotations = originalTypeAnnotations;
```

Comment:
It is better to create new map to store merged values

Reply:
We cannot do that, because we neeeds to merge them with original annotations